kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/dsl/NameableNamespaceableResource.java

```diff
@@ -0,0 +1,22 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.kubernetes.client.dsl;
+
+public interface NameableNamespaceableResource<T> extends Resource<T> {
+  Resource<T> withName(String name);
```

Comment:
This could be NamespaceableResource so that withName and inNamespace could be called in either order.